pandas/tools/tests/test_concat.py

```diff
@@ -27,6 +27,405 @@ def setUp(self):
         self.mixed_frame['foo'] = 'bar'
 
 
+class TestConcatAppendCommon(ConcatenateBase):
+
+    def test_concat_categorical(self):
+        # GH 13524
+
+        # same categories -> category
+        s1 = pd.Series([1, 2, np.nan], dtype='category')
+        s2 = pd.Series([2, 1, 2], dtype='category')
+
+        exp = pd.Series([1, 2, np.nan, 2, 1, 2], dtype='category')
+        tm.assert_series_equal(pd.concat([s1, s2], ignore_index=True), exp)
+        tm.assert_series_equal(s1.append(s2, ignore_index=True), exp)
+
+        # partially different categories => not-category
+        s1 = pd.Series([3, 2], dtype='category')
+        s2 = pd.Series([2, 1], dtype='category')
+
+        exp = pd.Series([3, 2, 2, 1])
+        tm.assert_series_equal(pd.concat([s1, s2], ignore_index=True), exp)
+        tm.assert_series_equal(s1.append(s2, ignore_index=True), exp)
+
+        # partially different categories with union_categoricals => category
+        exp = pd.Series(pd.Categorical([3, 2, 2, 1], categories=[2, 3, 1]))
+        tm.assert_series_equal(pd.concat([s1, s2], ignore_index=True,
+                                         union_categoricals=True), exp)
+        tm.assert_series_equal(s1.append(s2, ignore_index=True,
+                                         union_categoricals=True), exp)
+
+        # completelly different categories (same dtype) => not-category
+        s1 = pd.Series([10, 11, np.nan], dtype='category')
+        s2 = pd.Series([np.nan, 1, 3, 2], dtype='category')
+
+        exp = pd.Series([10, 11, np.nan, np.nan, 1, 3, 2])
+        tm.assert_series_equal(pd.concat([s1, s2], ignore_index=True), exp)
+        tm.assert_series_equal(s1.append(s2, ignore_index=True), exp)
+
+        # completelly different categories (same dtype) with union_categoricals
+        # => category
+        exp = pd.Series(pd.Categorical([10, 11, np.nan, np.nan, 1, 3, 2],
+                                       categories=[10, 11, 1, 2, 3]))
+        tm.assert_series_equal(pd.concat([s1, s2], ignore_index=True,
+                                         union_categoricals=True), exp)
+        tm.assert_series_equal(s1.append(s2, ignore_index=True,
+                                         union_categoricals=True), exp)
+
+    def test_concat_categorical_coercion(self):
+        # GH 13524
+
+        # values are all in category => category
+        s1 = pd.Series([1, 2, np.nan], dtype='category')
+        s2 = pd.Series([2, 1, 2])
+
+        exp = pd.Series([1, 2, np.nan, 2, 1, 2])
```

Comment:
Sorry, the comment must be changed because `category + non-category = non-category` is the principle. If `union_categoricals=True`, the result is `category` because non-category values are all in the categories.